StringOps.java

```diff
@@ -22,21 +22,112 @@ public class StringOps {
     //////                                               ///////
     ////////////////////////////////////////////////////////////
     public static void main(String[] args) {
-        
+        System.out.println(capVowelsLowRest("Hello World"));
+        camelCase("   Intro  to   coMPUter    sCIEncE ");
+        allIndexOf("Hello world",'l'); // output: {2, 3, 9}
     }
 
     public static String capVowelsLowRest (String string) {
-        // Write your code here:
-        return "";
+        String vowels = "aeiou";
+        String capVowels = "AEIOU";
+        String withBigVowels = "";
+        boolean isVowel = false;
+        for (int i = 0; i < string.length(); i++) {
+            for (int j = 0; j < vowels.length(); j++) {
+                if (string.charAt(i) == vowels.charAt(j)) {
+                    withBigVowels += (char) (string.charAt(i) - 32);
+                    isVowel = true;
+                } else if (string.charAt(i) == capVowels.charAt(j)) {
+                    withBigVowels += string.charAt(i);
+                    isVowel = true;
+                }
+            }
+            if ((string.charAt(i) >= 'A' && string.charAt(i) <= 'Z') && !isVowel) {
+                withBigVowels += (char) (string.charAt(i) + 32);
+            } else if (!isVowel) {
+                withBigVowels += string.charAt(i);
+            }
+            isVowel = false;
+        }
+        return withBigVowels;
     }
 
     public static String camelCase (String string) {
-        // Write your code here:
-        return "";
+        String camelCaseStr = "";
+        int j = 0;
+        boolean nextWord = false;
+        do {
+            if (string.charAt(j) == ' ') {
+                j++;
+            }
+        } while (j < string.length() && string.charAt(j) == ' ');
+        // Change first appearing letter to lowercase or if it's lowercase then add it
+        if (string.charAt(j) >= 'A' && string.charAt(j) <= 'Z') {
+                camelCaseStr += (char) (string.charAt(j) + 32);
+        } else if (string.charAt(j) >= 'a' && string.charAt(j) <= 'z') {
+            camelCaseStr += string.charAt(j);
+        }
```

Comment:
Since we assume there are only letters and spaces there is no need to specify what happens in the last else.
```suggestion
        if (string.charAt(j) >= 'A' && string.charAt(j) <= 'Z') {
                camelCaseStr += (char) (string.charAt(j) + 32);
        } else {
            camelCaseStr += string.charAt(j);
        }
```